### ISSUE TITLE
Maxwell unable to find DB when using init_position

### ISSUE THREAD

**Issue by Shukla-Ankur** [zendesk/maxwell#1849]
Mysql v8.0.23
Maxwell v1.33.0

We run maxwell with below config
`java -jar /home/ubuntu/maxwell/maxwell-1.33.0.jar --user=maxwell --password=maxwell --host=xxx --client_id=xxxx  --replica_server_id=xxxyyyy --producer=kafka --kafka.bootstrap.servers=xxxxxxx --kafka_topic=xxxxx --output_ddl=true --gtid_mode=true --producer_partition_by=column --producer_partition_columns=tenant_id --producer_partition_by_fallback=table --metrics_type=jmx --init_position=binlog.000137:4 --filter="exclude: *.*, include: testDb.*" &
`

Grants for maxwell
```
| GRANT SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO `maxwell`@`%` |
| GRANT ALL PRIVILEGES ON `maxwell`.* TO `maxwell`@`%`                        |
+--------------------------------------------------------
```

This means that replication user has access to required DBs. However, when I use init_position parameter, giving even the latest binlog file, I see following error 
```
java.lang.RuntimeException: Couldn't find database testDb
        at com.zendesk.maxwell.replication.TableCache.processEvent(TableCache.java:26) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getTransactionRows(BinlogConnectorReplicator.java:521) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:645) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:180) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:255) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183) ~[maxwell-1.33.0.jar:?]
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:286) ~[maxwell-1.33.0.jar:?]
```

This error goes away if init_position is not used and maxwell can start reading

**Comment by Shukla-Ankur** [comment]
I am connecting multiple maxwell instances to same Mysql machine. So the view should be correct only, since it is captured already when 1st maxwell started.
If init_position is not given, and initial binlog files are not available, how does maxwell capture consistent snapshot of DBs and tables? This is actually the case here because the issue resolves when init_position option is not given and earliest binlog files are no longer available.

**Comment by osheroff** [comment]
I'm a bit confused.  what's the first binlog position in the schemas table? 

```
mysql> select binlog_file, binlog_position from `schemas` order by id limit 1;
```

**Comment by Shukla-Ankur** [comment]
The first binlog position is
` mysql-bin-changelog.001392 |           20849`

Maxwell was reporting that it could not find tables. But I could see the "table-create" operation in schemas table.

**Comment by Shukla-Ankur** [comment]
any updates here @osheroff ??
We ran into this issue in production actually.

**Comment by osheroff** [comment]
> any updates here @osheroff ??

No, I'm just guessing in the dark with your setup.  If you gave me a way to reproduce the error I'd look at it but I can't tell what's going on otherwise.

> Let's say I have a mysql setup and binlog files are rotated on a daily basis. If I want to set up a new maxwell, how does maxwell determine the DB schemas if initial binlog files are no longer available?

It captures the current schema (at the end of the binlog).  It can't go back to the start of the binlogs.

> In production, when this erro came, I dropped the maxwell DB as last resort and restarted a new maxwell process with a init_position. It also failed with same error. 

Yeah, you can't do that.  There's no way for maxwell to recover the schema as of an arbitrary `init_position`.  I shoulda maybe document this better.

**Comment by mahipalkeizer** [comment]
I faced the same issue, In my case the issue was that when i created  the database maxwell instances were shut down so maxwell did not make an entry for newly created database in its schemas table. And when i made any query on newly created database maxwell read the query from binglog but could not find newly database in its schemas table. 

I was able to resolve it by `deleting all the contents of schemas table` or by `deleting maxwell DB`.
